src/ASM/ASMs2DmxLag.C

```diff
@@ -222,11 +222,11 @@ void ASMs2DmxLag::closeBoundaries (int dir, int, int)
 
 bool ASMs2DmxLag::getSize (int& n1, int& n2, int basis) const
 {
-  if (basis <= 1)
-    return this->ASMs2DLag::getSize(n1,n2,1);
+  if (basis < 1 || basis > static_cast<int>(nxx.size()))
+    return false;
 
-  n1 = nxx[basis-2];
-  n2 = nyx[basis-2];
+  n1 = nxx[basis-1];
+  n2 = nyx[basis-1];
```

Comment:
Similar fix for `ASMs3DmxLag::getSize()` ?

Reply:
kinda.